clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```
@@ -150,65 +163,50 @@ protected CompletableFuture<Void> internalSeek(
+        synchronized (subscriptions) {
+          if (subscriptions.isEmpty()) {
+            return;
           }
-        }
-
-        // Seek to end of push
-        if (currentVersion != maxVersion) {
-          // get current subscriptions and seek to endOfPush
-          Set<Integer> partitions = new HashSet<>();
-          for (PubSubTopicPartition partitionSubscription: subscriptions) {
-            partitions.add(partitionSubscription.getPartitionNumber());
+          int maxVersion = -1;
+          for (PubSubTopicPartition topicPartition: subscriptions) {
+            int version = Version.parseVersionFromVersionTopicName(topicPartition.getPubSubTopic().getName());
+            if (version >= maxVersion) {
+              maxVersion = version;
+            }
           }
-          try {
-            LOGGER.info(
-                "New Version detected!  Seeking consumer to version: " + currentVersion
-                    + " in VersionSwaDetectionThread: " + this.getId());
-            seekToEndOfPush(partitions).get();
-          } catch (InterruptedException | ExecutionException e) {
-            LOGGER.error(
-                "Seek to End of Push Failed for store: " + storeName + " partitions: " + partitions
-                    + " on VersionSwapDetectionThread: " + this.getId() + "will retry...",
-                e);
+
+          // Seek to end of push
+          if (currentVersion != maxVersion) {
+            // get current subscriptions and seek to endOfPush
+            for (PubSubTopicPartition partitionSubscription: subscriptions) {
+              partitions.add(partitionSubscription.getPartitionNumber());
+            }
           }
+
+          LOGGER.info(
+              "New Version detected!  Seeking consumer to version: " + currentVersion + " in consumer: "
+                  + changelogClientConfig.getConsumerName());
+          seekToEndOfPush(partitions).get();
         }
+      } catch (InterruptedException | ExecutionException e) {
```

can we catch all exception here?

---

Sure!